integreat_cms/core/signals/hix_signals.py

```diff
@@ -28,9 +28,10 @@ def page_translation_save_handler(instance, **kwargs):
         and instance.latest_version.content == instance.content
         and instance.latest_version.hix_score is not None
     )
+    if instance.hix_score:
+        return instance.hix_score
```

Comment:
> I modified page_translation_save_handler so it returns the existing HIX value if such extsts instead of skipping the process. HIX value must be now restored too. I hope this change doesn't bring any side effect to HIX process.

Yes this makes sense, but it would be better to refactor the whole function in my opinion, since the whole if cases are nested more than they need to be.

What do you think about e.g. the following:

```
    if identical_version := instance.all_versions.filter(
        content=instance.content, hix_score__isnull=False
    ).first():
        logger.debug(
            "Content of %r is identical to %r, copying HIX score %r",
            instance,
            identical_version,
            identical_version.hix_score,
        )
        instance.hix_score = identical_version.hix_score
        return

    if instance.hix_ignore or not instance.hix_enabled or not instance.content.strip():
        logger.debug(
            "HIX calculation pre save signal skipped for %r (ignored=%s, enabled=%s, empty=%s)",
            instance,
            instance.hix_ignore,
            instance.hix_enabled,
            not bool(instance.content.strip()),
        )
        instance.hix_score = None
        return
```

Reply:
Looks cool 😄 Applied!